src/database/remove-helper.js

```diff
@@ -0,0 +1,125 @@
+"use strict";
+
+var api = require("../api");
+var FirebaseError = require("../error");
+var request = require("request");
+var responseToError = require("../responseToError");
+var utils = require("../utils");
+const logger = require("../logger");
+
+class DatabaseRemoveHelper {
```

Comment:
I don't think you're getting a lot of value out of having these helpers as a class. I'd go with one of three options:

1. Have these just be helper methods that take `instance` as a first argument exported from this file that are reused in the remover class.
2. Inline these helper methods in `remove.js` as unexported functions.
3. Have these be methods on the main `DatabaseRemove` class, possibly prefixed with an underscore if you want to indicate that they're private.

Reply:
Those serve as a API level abstraction for RTDB, so that I can write a simple mock for it, and do some sanity check to make sure the promise actually get resolved eventually.

I don't like the name `DatabaseRemoveHelper` either. 
Do you have a suggestion for naming?